tests/test_twisted.py

```diff
@@ -201,11 +202,14 @@ def test_fakeDHParameterSupport(self, monkeypatch, allFile, recwarn):
         assert ctxFactory.ctxFactory is fakeCtxFactory
         assert "dhParameters" not in recorder.calls[0].kwargs
 
-    def test_realDHParameterSupport(self, monkeypatch, allFile):
+    def test_realDHParameterSupport(self, monkeypatch, tmpdir):
         """
         Pass DH parameters directly to CertificateOptions if the installed
```

Comment:
I reworked this so there are now 4 tests covering each of the {explict DH params, file DH params} x {no Twisted support, Twisted support} cases. There's a fair amount of duplication between the tests, hopefully that's okay?